src/api/app/views/webui/users/notifications/index.html.haml

```diff
@@ -43,8 +43,13 @@
                     %span.badge.badge-secondary= notification.notification_badge
                     %small.text-muted= time_ago_in_words(notification.created_at)
                   = link_to(n.title, notification.link_to_notification_target, class: 'text-word-break-all')
-                - unless params[:type] == 'done'
+                - unless params[:type] == 'read'
                   .actions.align-self-end.align-self-sm-start.pl-3.pt-3.pt-sm-0
                     = link_to(my_notification_path(id: notification),
-                                      method: :put, class: 'btn btn-sm btn-outline-success px-4', title: 'Mark as "Done"') do
+                                      method: :put, class: 'btn btn-sm btn-outline-success px-4', title: 'Mark as "Read"') do
                       %i.fas.fa-check
+                - unless notification.unread?
+                  .actions.align-self-end.align-self-sm-start.pl-3.pt-3.pt-sm-0
+                    = link_to(my_notification_path(id: notification),
+                                      method: :put, class: 'btn btn-sm btn-outline-danger px-4', title: 'Mark as "Unread"') do
```

Comment:
This isn't a dangerous thing to do right? Looks strange, very much like removal.

Reply:
How about the [undo](https://fontawesome.com/icons/undo?style=solid) icon?